Dockerfile

```diff
@@ -0,0 +1,40 @@
+ARG KUBECTL_VERSION=1.31.1
+ARG YQ_VERSION=v4.44.3
+
+# Stage 1: Download binaries
+FROM alpine as download_binary
+
+ARG KUBECTL_VERSION
+ARG YQ_VERSION
+
+# Install curl and certificates, and clean up in one layer to reduce image size
+RUN apk update && apk add --no-cache \
+    curl \
+    ca-certificates
+
+# Download kubectl binary
+RUN curl -sSL -o /kubectl "https://dl.k8s.io/release/v${KUBECTL_VERSION}/bin/linux/amd64/kubectl" \
+  && chmod +x /kubectl
+
+# Download yq binary
+RUN curl -sSL -o /yq "https://github.com/mikefarah/yq/releases/download/${YQ_VERSION}/yq_linux_amd64" \
+  && chmod +x /yq
+
+# Stage 2
+FROM alpine 
+
+RUN apk update && apk add --no-cache \
+  sudo \
+  && adduser -D -s /bin/bash slice \
+  && echo 'slice ALL=(ALL) NOPASSWD:ALL' > /etc/sudoers.d/slice \
+  && chmod 0440 /etc/sudoers.d/slice 
+
+# Copy binaries from the download_binary stage
+COPY --from=download_binary /kubectl /usr/local/bin/kubectl
+COPY --from=download_binary /yq /usr/local/bin/yq
+COPY --from=download_binary /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/ca-certificates.crt
+
+# Copy kubectl-slice from local filesystem
+COPY kubectl-slice /usr/local/bin/kubectl-slice
+USER slice
+WORKDIR /home/slice
```

Comment:
(Moving the message from the other thread)

You seem to have changed this to Alpine. For Go, this has a few complications.

Alpine ships with a syscall library called `musl`. We compile on an ubuntu OS, which ships with a syscall library called `libc`. Both are not completely compatible with each other and while we can statically link the binary I would personally prefer keeping them apples-to-apples to reduce the risk situation.

People would still be able to install `kubectl-slice` on an Alpine machine, but I don't want to give the impression that the container support would be just Alpine.

I know Alpine images are a bit on the heavier side, but a) this app won't run as a server, but as part of the CI; and b) the base image being debian means that there's a high chance the heaviest part of the image, weighting `75 MB`, already exists in the host running this. 

`kubectl` alone is ~56 MB on Linux, so I don't know if I would be too worried about container size but more about usability.

Reply:
I guess lastly, too, Alpine ships with `ash`, not with `bash`. Which means that people might run into issues if they didn't pull the appropriate shebang on top of their files and/or how they run commands.

Reply:
Thank you for the detailed feedback 🙌 ! I understand the concerns around musl vs. libc syscall compatibility, Sticking with Debian does seem like a safer option to ensure compatibility and reduce risk. I agree that while Alpine images offer a smaller footprint, the size difference here likely won’t be as impactful since the app will run in CI, not as a server, and kubectl itself is quite large.

i have updated the base image to Debian to maintain consistency and avoid potential issues :)